src/message.ts

```diff
@@ -0,0 +1,93 @@
+import { IAddress, IMessage } from "./interface";
+import { DEFAULT_MESSAGE_VERSION, MESSAGE_PREFIX } from "./constants";
+import { Address } from "./address";
+
+const createKeccakHash = require("keccak");
+
+export class Message {
+    /**
+     * Actual message being signed.
+     */
+    public data: Uint8Array;
+    /**
+     * The message signature.
+     */
+    public signature?: Uint8Array;
+    /**
+     * Address of the wallet that performed the signing operation.
+     */
+    public address?: IAddress;
+    /**
+     * Number representing the message version.
+     */
+    public version?: number;
+
+    constructor(options: { data: Uint8Array; signature?: Uint8Array; address?: IAddress; version?: number }) {
+        this.data = options.data;
+        this.signature = options.signature;
+        this.address = options.address;
+        this.version = options.version ? options.version : DEFAULT_MESSAGE_VERSION;
+    }
+}
+
+export class MessageComputer {
+    constructor() {}
+
+    computeBytesForSigning(message: IMessage): Uint8Array {
+        const messageSize = Buffer.from(message.data.length.toString());
+        const signableMessage = Buffer.concat([messageSize, message.data]);
+        let bytesToHash = Buffer.concat([Buffer.from(MESSAGE_PREFIX), signableMessage]);
+
+        return createKeccakHash("keccak256").update(bytesToHash).digest();
+    }
+
+    computeBytesForVerifying(message: IMessage): Uint8Array {
+        return this.computeBytesForSigning(message);
+    }
+
+    packMessage(message: IMessage): {
+        message: string;
+        signature: string;
+        address: string;
+        version: number;
+    } {
+        return {
+            message: Buffer.from(message.data).toString("hex"),
+            signature: message.signature ? Buffer.from(message.signature).toString("hex") : "",
+            address: message.address ? message.address.bech32() : "",
+            version: message.version ? message.version : DEFAULT_MESSAGE_VERSION,
+        };
+    }
+
+    unpackMessage(packedMessage: { message: string; signature?: string; address?: string; version?: number }): Message {
+        let dataHex = this.trimHexPrefix(packedMessage.message);
+        const data = Buffer.from(dataHex, "hex");
+
+        let signature: Uint8Array | undefined = undefined;
+        if (packedMessage.signature) {
+            let signatureHex = this.trimHexPrefix(packedMessage.signature);
+            signature = Buffer.from(signatureHex, "hex");
+        }
```

Comment:
Can also be (proposal):

```
const signatureHex = this.trimHexPrefix(packedMessage.signature || "");
const signature = Buffer.from(signatureHex, "hex")
```

Will be an empty buffer if signature not provided, but that is maybe even better.

Reply:
Replaced with your proposal. Thanks!